publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java

```diff
@@ -337,10 +339,18 @@ public void refresh(SortableIdentifier identifier) {
         }
     }
 
-    public Stream<TicketEntry> fetchAllTicketsForElevatedUser(UserInstance userInstance,
-                                                              SortableIdentifier publicationIdentifier)
-        throws NotFoundException {
-        var resource = fetchResourceForElevatedUser(userInstance.getCustomerId(), publicationIdentifier);
+    public Stream<TicketEntry> fetchAllTicketsForUser(UserInstance userInstance,
+                                                      SortableIdentifier publicationIdentifier)
+        throws NotFoundException, ForbiddenException {
+        var resource = getResourceByIdentifier(publicationIdentifier);
+
+        var permissionStrategy = PublicationPermissionStrategy.create(resource.toPublication(), userInstance,
+                                                                      uriRetriever);
+
+        if (!permissionStrategy.allowsAction(PublicationOperation.UPDATE)) {
```

Comment:
Do this let contributors see all tickets for a publication too? Is that ok? I am not sure, so I ask to make sure it has been considered.

Reply:
I would move this validation out of `ResourceService` to handler, because now 
fetchAllTicketsForUser() does not return all tickets for user if you are not right user. I think it makes it clear when validation is lying on the top layer + single responsibility/ separation of concerns .

Reply:
I agree! refactored it now 👍